src/lightning/fabric/strategies/fsdp.py

```diff
@@ -711,7 +711,10 @@ def _apply_optimizers_during_fsdp_backward(
     assert param_handles, f"Module {module} does not appear to contain any FSDP modules."
     fsdp_state = _get_module_fsdp_state(module)
     assert fsdp_state is not None
-    fsdp_stream = fsdp_state._streams["post_backward"]
+
+    # There isn't a formal contract on the stream logic for FSDP, so we unfortunately
+    # have to manually bridge internal refactors. (e.g. https://github.com/pytorch/pytorch/pull/103902)
+    fsdp_stream = getattr(fsdp_state, "_streams_post_backward", None) or fsdp_state._streams["post_backward"]
```

Comment:
You can check the current pytorch version by importing the variable from https://github.com/Lightning-AI/lightning/blob/master/src/lightning/fabric/utilities/imports.py. Doing this has the advantage that we know what to remove exactly as the support window is updated in the future

Reply:
IMO we could just support torch > 2.0 only. This attribute name has already been refactored several times :)